File: Reshape-the-Matrix.py
class Solution:
    def matrixReshape(self, mat, r, c):
        m = len(mat)
        n = len(mat[0])
        if m * n != r * c: return mat

        result = [[0] * c for _ in range(r)]
        row, col = 0, 0

        for i in range(m):
            for j in range(n):
                result[row][col] = mat[i][j]
                col += 1
                if col == c:
                    col = 0
                    row += 1
        return result

File: Reshape-the-Matrix.js
var matrixReshape = function(mat, r, c) {
    let m = mat.length, n = mat[0].length;
    if (m * n !== r * c) return mat;

    let result = Array.from({ length: r }, () => Array(c).fill(0));
    let row = 0, col = 0;

    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            result[row][col] = mat[i][j];
            col++;
            if (col === c) {
                col = 0;
                row++;
            }
        }
    }
    return result;
};
